app/workers/argo_sync_worker.rb

```diff
@@ -0,0 +1,66 @@
+class ArgoSyncWorker
+  include Sidekiq::Worker
+  sidekiq_options retry: false
+
+  def perform(helm_infrastructure_id)
+    infrastructure = HelmInfrastructure.find(helm_infrastructure_id)
```

Comment:
please validate if the `helm_infrastructure_id` couldn't be found

Reply:
updated